gcloud-java-compute/src/main/java/com/google/gcloud/compute/ComputeImpl.java

```diff
@@ -684,6 +743,187 @@ public Boolean call() {
     }
   }
 
+  @Override
+  public Address get(final AddressId addressId, AddressOption... options) {
+    final Map<ComputeRpc.Option, ?> optionsMap = optionMap(options);
+    try {
+      com.google.api.services.compute.model.Address answer =
+          runWithRetries(new Callable<com.google.api.services.compute.model.Address>() {
+            @Override
+            public com.google.api.services.compute.model.Address call() {
+              switch (addressId.type()) {
+                case REGION:
+                  RegionAddressId regionAddressId = (RegionAddressId) addressId;
```

Comment:
I think I didn't look carefully of what we did before with operation, and this is following the same pattern.
Adding the type is a step forward but type does not force (or even specify) what implementation of the interface is expected. Maybe instead (or in addition) we should use interface inheritance to add the specific getters so we will not need to cast or expect a specific implementation or ours when input API is open.

This does not fix the `setProject` issue which could be fixed in various ways as suggested above (or by adding to the specific implementation a copy-constcutor based on its matching interface.